.github/workflows/ci.yaml

```diff
@@ -17,10 +17,10 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
 
       - name: Setup Go environment
-        uses: actions/setup-go@v2.1.3
+        uses: actions/setup-go@93397bea11091df50f3d7e59dc26a7711a8bcfbe # v4.1.0
```

Comment:
Why are we using hashes instead of versions? Does `v4.1.0` not work?

Also, is there a solid reason to not just use `actions/setup-go@v4`, and get bugfixes, etc automatically?

Reply:
IMO, for security reasons, it is a good practice to use a hash here (https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions)

Reply:
True, but `actions/setup-go` is a first-party action (owned by GitHub itself)

Reply:
The general idea is to use renovate in the near future to take care and keep up-to-date the github actions. I'm pinging @kim-tsao also as the author of the EPIC (the hash was added as part of the acceptance criteria)

Reply:
We are following Red Hat's GitHub Security Guidelines and this is one of the best practices.  Yes, we can make an exception for trusted actions but then we'd need to keep track of what's trusted and what's not trusted.  If we adopt a blanket requirement for hashes, we can also prevent the use of any future untrusted actions without having to re-scan our repos.  As @thepetk mentioned, renovate should handle updating the actions in the future so this is just a one time manual change.